DTAConfig/Settings/FileSourceDestinationInfo.cs

```diff
@@ -100,17 +101,7 @@ public void Apply()
                     break;
 
                 case FileOperationOptions.KeepChanges:
-                    if (!File.Exists(DestinationPath))
-                    {
-                        if (File.Exists(CachedPath))
-                            File.Copy(CachedPath, DestinationPath, false);
-                        else
-                            File.Copy(SourcePath, DestinationPath, false);
-                    }
-
-                    Directory.CreateDirectory(Path.GetDirectoryName(CachedPath));
-                    File.Copy(DestinationPath, CachedPath, true);
-
+                    CreateLinkToFile(sourcePath, destinationPath);
```

Comment:
Don't forget fallback to regular copying

Reply:
> Don't forget fallback to regular copying

Why though? This implementation doesn't need that undefined behavior, IMO.

If you want, i can revert changes for `KeepChanges` to original file copy and add new `ImmutableLink` option for links.

Reply:
with fallback -- you are proposing a feature that at least fallback to previous behavior. linking is an "optimization" to copying.

without fallback -- you are breaking the client for partial users

Not all file systems support linking. An "ImmutableLink" option is also not a good idea because nor us developer or the modder should decide whether all users must use linking or not.

Reply:
> An "ImmutableLink" option is also not a good idea because nor us developer or the modder should decide whether all users must use linking or not.

Why not though? Linking is a different behavior from copying.

Reply:
> > An "ImmutableLink" option is also not a good idea because nor us developer or the modder should decide whether all users must use linking or not.
> 
> Why not though? Linking is a different behavior from copying.

Linking without a fallback succeeds on partial users and fails on other users.

Reply:
> > An "ImmutableLink" option is also not a good idea because nor us developer or the modder should decide whether all users must use linking or not.
> 
> Why not though? Linking is a different behavior from copying.

Also, current file operation options includes: overwrite on mismatch, don't overwrite, etc. These are different behaviors about *whether* to copy. If we want to differ linking from copying, we need to add a new group of options. e.g., overwrite on mismatch with copying, overwrite on mismatch with linking